src/Arcus.Messaging.Pumps.ServiceBus/Configuration/ServiceBusMessagePumpOptions.cs

```diff
@@ -152,4 +158,27 @@ public TimeSpan SessionIdleTimeout
             }
         }
     }
+
+    /// <summary>
+    /// Represents the options related to the <see cref="ServiceBusMessagePumpOptions.Hooks"/> of the registered Azure Service Bus message pump.
+    /// </summary>
+    public class ServiceBusMessagePumpHooksOptions
+    {
+        internal Func<IServiceProvider, Task> BeforeStartupAsync { get; set; } = _ => Task.CompletedTask;
+
+        /// <summary>
+        /// Sets a function on the message pump that should run before the pump receives Azure Service Bus messages.
+        /// Useful for when dependent systems are not always directly available.
+        /// </summary>
+        /// <remarks>
+        ///     ⚠️ Multiple calls will override each other.
```

Comment:
Wouldn't it be nice if multiple calls would result in multiple hooks being executed ?
It would make it all more complex of course ....

Reply:
Yes, would they need to be run serial, parallel, should only one be finished or all...

Reply:
Indeed, let's keep it as is.